### ISSUE TITLE
Save function and load_script_textdomain depends on editing admin user language

### ISSUE THREAD

**Issue by dorkster100** [WordPress/gutenberg#22983]
Let's say i have a German website, But my admin user is an English speaking person, if his account is not german, then the script language JSON that is loaded is in English. So the `save()` function of the gutenberg block will translate strings in English and save them to DB.

Should i not be able to do this in the custom block?

`save: ( props ) => {
			return (
				<div>
					<h2>{ __('Open', 'dummy-domain') }</h2>
				</div>
			);
		},`

**Expected behavior**
Display in admin area strings in admin language but save those strings in website language.

I sadly tested this only on WP 5.3.3, but i checked the i10n.php of the 5.4.1 and it seems the same. Sorry if this is already fixed.

**Comment by gziolo** [comment]
I was exploring how this type of use case could eorj in #23268. No success so far though.

**Comment by dorkster100** [comment]
With server-side rendering it could certainly work, however for now to me this seems like a bug and, the function that `load_script_textdomain` uses to `determine_locale` should not have these checks:
`
if ( is_admin() ) {
	$determined_locale = get_user_locale();
}

if ( isset( $_GET['_locale'] ) && 'user' === $_GET['_locale'] && wp_is_json_request() ) {
	$determined_locale = get_user_locale();
}
`

Maybe it would be better to use `get_locale` instead? It would of course be better to be able to see in user's language but save in the website's, but i would consider this as a feature and improvement?

**Comment by youknowriad** [comment]
I don't think we ever supported translation in save functions. This is basically related to the Multi-lingual phase of Gutenberg because it's about translating content (save functions are content). So I'm going to close this issue and refer to that bigger scheme instead but for now, a dynamic behavior on top of the static one is the right approach.
